src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -354,9 +357,9 @@ protected NestedResourceInfoScope ParentNestedResourceInfoScope
 
                     // Get the resourceSet's parent (if any)

```

Comment:
If the above assert is true, then we should have a ParentOfParent should not be null. Fixed comment.

---
In reply to: [127298267](https://github.com/OData/odata.net/pull/894#discussion_r127298267) [](ancestors = 127298267)